### PR TITLE
0.1.0

### PR DESCRIPTION
**Pull request by jkuri**

### CODE REVIEW

File: src/api/process-manager.ts [hunk 1]
```diff
@@ -1,129 +1,167 @@
...
-              exitProcess(uuid);
-            });
-          }
-        });
+export let jobProcesses: JobProcess[] = [];
 
-        resolve(proc);
-      });
+export function startBuild(): Promise<null> {
+  let gitUrl = 'https://github.com/jkuri/bterm.git';
```
Comment (irmana) [inline src/api/process-manager.ts]: Why don't you put this into the config? Also, isn't our gitURL `https://github.com/bleenco/bterm.git`?

File: src/api/process-manager.ts [hunk 2]
```diff
@@ -1,129 +1,167 @@
...
 }
 
-function generateId(): string {
-  return uuid();
+export function findDockerImageBuildJob(name: string): JobProcess | null {
+  const index = jobProcesses.findIndex(job => job.image_name && job.image_name === name);
+  return jobProcesses[index] || null;
 }
+
+// export function getRunningJob(id: number): Observable<any> {
```
Comment (irmana) [inline src/api/process-manager.ts]: Do we need this commented code or can we remove it?

File: src/api/process.ts [hunk 3]
```diff
@@ -1,42 +1,73 @@
...
-  return proc;
+  return job;
 }
 
-export function exitProcess(id: string): void {
-  const index = processes.findIndex(proc => proc.id === id);
-  if (index === -1) {
-    return;
-  }
+// export function exitProcess(id: number): void {
```
Comment (irmana) [inline src/api/process.ts]: Cleanup?

Comment (jkuri) [inline src/api/process.ts]: sure, its under heavy WIP :)

File: src/api/socket.ts [hunk 4]
```diff
@@ -40,39 +39,57 @@ export class SocketServer {
...
+                  conn.next({ type: 'terminalOutput', data: event });
+                }, err => {
+                  console.error(err);
+                }, () => {
+                  console.log('Docker image build completed.');
+                });
               break;
             }
 
+            // switch (event.type) {
```
Comment (irmana) [inline src/api/socket.ts]: Cleanup...?

File: src/api/process.ts (hunk 3 above)
Comment (jkuri) [inline src/api/process.ts]: same stands true for both previous comments.

**Pull request merged**